### ISSUE TITLE
allow selecting a custom branch/ref/tag for dependencies from git/github when using mix igniter.install

### ISSUE THREAD

**Issue by albinkc** [ash-project/igniter#84]
I can do this with deps in mix.exs currently, could be a nice-to-have in igniter.install

Reference: https://hexdocs.pm/mix/1.7.0/Mix.Tasks.Deps.html

**Comment by zachdaniel** [comment]
Yep! Agreed. The main thing we need is syntax for it. Right now you can say

`--install foo@github:ash-project/igniter`
and we need a way to separate the branch. Maybe just `:`, like `--install foo@github:ash-project/igniter:main`

**Comment by mathieupost** [comment]
As a workaround you can specify the dependency with repo and branch override already in your `mix.exs` like:
```
{:ash_authentication, [github: "team-alembic/ash_authentication", branch: "igniter-installer"]}
```

Then, when running `mix igniter.install ash_authentication` it will ask this and you can just say no:
```
Dependency ash_authentication is already in mix.exs. Should we replace it?

Desired: `{:ash_authentication, "~> 4.0"}`
Found: `{:ash_authentication, [github: "team-alembic/ash_authentication", branch: "igniter-installer"]}`
 [Yn] n
```

**Comment by zachdaniel** [comment]
That should be fine, the first @ would split package and version, and any other `@` can be considered part of its segment

**Comment by tomekowal** [comment]
So, in case of:
```
package@git:git@github.com:organization/package.git@v0.69.3
```
we would basically:
1. Split in to {package, rest}
2. Split rest into {url, version} by last `@`
If that is what you mean, I can make a PR :)

**Comment by zachdaniel** [comment]
Yep, one of the two! Probably parsing the URI would be good.

**Comment by djgoku** [comment]
This work was completed with this commit: https://github.com/ash-project/igniter/commit/05422953e00bee555bbc6ac8eda86650b4592639
